src/fileimpl.cpp

```diff
@@ -590,29 +590,41 @@ std::string pseudoTitle(const Dirent& d)
   return std::string(1, d.getNamespace()) + '/' + d.getTitle();
 }
 
+bool checkIndirectDirentAccessor(const IndirectDirentAccessor& accessor, entry_index_type totalCount) {
+  const entry_index_type direntCount = accessor.getDirentCount().v;
+  std::shared_ptr<const Dirent> prevDirent;
+  for ( entry_index_type i = 0; i < direntCount; ++i ) {
+    if (accessor.getDirectIndex(title_index_t(i)).v >= totalCount) {
+      std::cerr << "Invalid title index entry" << std::endl;
+      return false;
+    }
+
+    const std::shared_ptr<const Dirent> dirent = accessor.getDirent(title_index_t(i));
+    if ( prevDirent && !(pseudoTitle(*prevDirent) <= pseudoTitle(*dirent)) ) {
+      std::cerr << "Title index is not properly sorted:\n"
+                << "  #" << i-1 << ": " << pseudoTitle(*prevDirent) << "\n"
+                << "  #" << i   << ": " << pseudoTitle(*dirent) << std::endl;
+      return false;
+    }
+    prevDirent = dirent;
+  }
+  return true;
+}
```

Comment:
If this function detects and reports a violation, it's not clear which title index contains that issue. If that's not important, then this check could output a generic error message, without trying to pinpoint the error more closely.

Reply:
Yes, it will be complicated to have a real useful information here.
And if we have a invalid zim file it is probably a bug on the creator side. User running zimcheck/zimwriterfs will probably not be able to fix this.
Only debug made by dev will fix it.

So we can use a generic message here.